ingenialink/dictionary.py

```diff
@@ -1,23 +1,150 @@
+import xml.etree.ElementTree as ET
+
 from abc import ABC, abstractmethod
+from ingenialink.register import REG_DTYPE, REG_ACCESS
+from ingenialink import exceptions as exc
+from enum import Enum
+
+
+# Dictionary constants guide:
+# Each constant has this structure: DICT_ORIGIN_END
+# ORIGIN: The start point of the path
+# END: The end point of the path
+# ORIGIN: ROOT
+DICT_ROOT = "."
+DICT_ROOT_HEADER = f"{DICT_ROOT}/Header"
+DICT_ROOT_VERSION = f"{DICT_ROOT_HEADER}/Version"
+DICT_ROOT_BODY = f"{DICT_ROOT}/Body"
+DICT_ROOT_DEVICE = f"{DICT_ROOT_BODY}/Device"
+DICT_ROOT_CATEGORIES = f"{DICT_ROOT_DEVICE}/Categories"
+DICT_ROOT_CATEGORY = f"{DICT_ROOT_CATEGORIES}/Category"
+DICT_ROOT_ERRORS = f"{DICT_ROOT_BODY}/Errors"
+DICT_ROOT_ERROR = f"{DICT_ROOT_ERRORS}/Error"
+DICT_ROOT_AXES = f"{DICT_ROOT_DEVICE}/Axes"
+DICT_ROOT_AXIS = f"{DICT_ROOT_AXES}/Axis"
+DICT_ROOT_REGISTERS = f"{DICT_ROOT_DEVICE}/Registers"
+DICT_ROOT_REGISTER = f"{DICT_ROOT_REGISTERS}/Register"
+# ORIGIN: REGISTERS
+DICT_REGISTERS = "./Registers"
+DICT_REGISTERS_REGISTER = f"{DICT_REGISTERS}/Register"
+# ORIGIN: LABELS
+DICT_LABELS = "./Labels"
+DICT_LABELS_LABEL = f"{DICT_LABELS}/Label"
+# ORIGIN: RANGE
+DICT_RANGE = "./Range"
+# ORIGIN: ENUMERATIONS
+DICT_ENUMERATIONS = "./Enumerations"
+DICT_ENUMERATIONS_ENUMERATION = f"{DICT_ENUMERATIONS}/Enum"
+
+dtype_xdf_options = {
+            "float": REG_DTYPE.FLOAT,
+            "s8": REG_DTYPE.S8,
+            "u8": REG_DTYPE.U8,
+            "s16": REG_DTYPE.S16,
+            "u16": REG_DTYPE.U16,
+            "s32": REG_DTYPE.S32,
+            "u32": REG_DTYPE.U32,
+            "s64": REG_DTYPE.S64,
+            "u64": REG_DTYPE.U64,
+            "str": REG_DTYPE.STR
+        }
 
+access_xdf_options = {
+            "r": REG_ACCESS.RO,
+            "w": REG_ACCESS.WO,
+            "rw": REG_ACCESS.RW
+        }
 
-class Categories(ABC):
-    """Categories Abstract Base Class.
+
+# Attributes of a register
+class AttrRegDict:
```

Comment:
Add this constant inside the dictionary class, it will be better for inheritance.